### PR TITLE
Feature rhino turn with encoder

### PR DESCRIPTION
**Pull request by crisvdn**
This pull request implements a working rhino class. This rhino class can turn with the encoders and drive forward/ backwards.

### CODE REVIEW

File: code/main.cpp [hunk 1]
```diff
@@ -1,35 +1,54 @@
...
+    // Create qik_controller for robot
     r2d2::moving_platform::qik_2s12v10_c qik =
         r2d2::moving_platform::qik_2s12v10_c(usart, qik_2s12v10_reset_pin);
 
-    // Create internalcommunication
+    // create internal commmunication
     r2d2::comm_c comm;
 
-    // Create The moving platform the Beetle
-    auto beetle =
```
Comment (Joepieler) [inline code/main.cpp:26]: a qik can't be a beetle and a rhino at the same time

File: code/main.cpp [hunk 2]
```diff
@@ -1,35 +1,54 @@
...
+        while (1) {
+            rhino.process();
+            hwlib::wait_ms(100);
+        }
+    }
 
-    while (true) {
-        beetle.process();
-        hwlib::wait_ms(100);
+    if (rhino_motor_test) {
```
Comment (Joepieler) [inline code/main.cpp]: No tests in the main of the release.

File: code/src/rhino.cpp [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+
+    void rhino_c::turn(int8_t degrees) {
+
+        // The puls. Its starts low.
+        bool low_m0 = false;
+        int counter_m0 = 0;
+        bool low_m1 = false;
+        int counter_m1 = 0;
+        // The adc input. is between 3000 and 3800.
+        unsigned int adc_voltage = 3500;
```
Comment (Joepieler) [inline code/src/rhino.cpp]: adc_voltage can be constexpr

File: code/src/rhino.cpp [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+                if (low_m1 == true) {
+                    counter_m1++;
+                }
+                low_m1 = false;
+            } else {
+                low_m1 = true;
+            }
+
+            if (counter_m0 ==
+                (int(encode_1_full_turn * turn_factor / 360 * degrees))) {
```
Comment (Joepieler) [inline code/src/rhino.cpp]: can be calculated before the loop

File: code/src/rhino.cpp [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+        if (degrees < 0) {
+            // turn right
+            turn_speed = turn_speed * -1;
+            degrees = degrees * -1;
+        }
+        // motor tests strats motors.
+        qik_2s12v10_motorcontroller.set_m0_speed(turn_speed);
+        qik_2s12v10_motorcontroller.set_m1_speed(-1 * turn_speed);
+
+        while (true && degrees != 0) {
```
Comment (Joepieler) [inline code/src/rhino.cpp]: the True can go away.

**Review by Joepieler — CHANGES_REQUESTED** [review]

File: code/src/rhino.cpp (hunk 5 above)
Comment (crisvdn) [reply code/src/rhino.cpp]: This has been changed.

File: code/src/rhino.cpp (hunk 4 above)
Comment (crisvdn) [reply code/src/rhino.cpp]: This calculation is now stored in a variable.

File: code/src/rhino.cpp (hunk 3 above)
Comment (crisvdn) [reply code/src/rhino.cpp]: This variable is now constexpr static.

File: code/main.cpp (hunk 2 above)
Comment (crisvdn) [reply code/main.cpp]: Fixed.

File: code/main.cpp (hunk 1 above)
Comment (crisvdn) [reply code/main.cpp:26]: Fixed.

**Review by Joepieler — APPROVED** [review]
Everything i asked for has been changed.

**Pull request merged**